sharedresource/v1alpha1/types_share.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright The OpenShift authors.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Share is the Schema for the shares API
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:subresource:status
+// +openshift:compatibility-gen:level=4
+type Share struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   ShareSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status ShareStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShareList contains a list of Share
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:subresource:status
+// +openshift:compatibility-gen:level=4
+type ShareList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []Share `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// ShareSpec defines the desired state of Share
+type ShareSpec struct {
+	// BackingResource captures the ConfigMap or Secret that is shared.
```

Comment:
Describe what adding a reference here will do.  What does it enable?

Reply:
You may already have something in flight @coreydaley, but feeling culpable since I originally crafted this, I'll offer an attempt at addressing @deads2k request:

`BackingResource captures the information needed to look up a ConfigMap or Secret stored in a set of namespaces available on the cluster.  Then, when any Pod which specifies a CSI Volume of the type supported by the corresponding driver and controller for Shares, the contents of the ConfigMap or Secret referenced here will be mounted in the Pod.`

Reply:
Why not just `ResourceReference` to make it similar to other references across all of our API, then also the comments from David about the shape of the reference struct would be in-line with other reference types we have.

Reply:
updated in next push